pkg/controllers/terminating/v1alpha1/terminate.go

```diff
@@ -38,105 +38,62 @@ type Terminator struct {
 	coreV1Client  corev1.CoreV1Interface
 }
 
-// cordonNodes takes in a list of expired nodes as input and cordons them
-func (t *Terminator) cordonNodes(ctx context.Context) error {
-	// 1. Get terminable nodes
-	nodeList, err := t.getLabeledNodes(ctx, provisioning.ProvisionerTerminablePhase)
-	if err != nil {
-		return err
-	}
-	// 2. Cordon nodes
-	for _, node := range nodeList {
-		persisted := node.DeepCopy()
-		node.Spec.Unschedulable = true
-		node.Labels = functional.UnionStringMaps(
-			node.Labels,
-			map[string]string{provisioning.ProvisionerPhaseLabel: provisioning.ProvisionerDrainingPhase},
-		)
-		if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
-			return fmt.Errorf("patching node %s, %w", node.Name, err)
-		}
-		zap.S().Debugf("Cordoned node %s", node.Name)
+// cordonNode cordons a node
+func (t *Terminator) cordonNode(ctx context.Context, node *v1.Node) error {
+	persisted := node.DeepCopy()
+	node.Spec.Unschedulable = true
+	if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
+		return fmt.Errorf("patching node %s, %w", node.Name, err)
 	}
+	zap.S().Debugf("Cordoned node %s", node.Name)
 	return nil
 }
 
-// terminateNodes takes in a list of expired non-drained nodes and calls drain on them
-func (t *Terminator) terminateNodes(ctx context.Context) error {
-	// 1. Get draining nodes
-	draining, err := t.getLabeledNodes(ctx, provisioning.ProvisionerDrainingPhase)
+// drainNode evicts pods from the node and returns true when fully drained
+func (t *Terminator) drainNode(ctx context.Context, node *v1.Node) (bool, error) {
+	// 1. Get pods on node
+	pods, err := t.getPods(ctx, node)
```

Comment:
We actually don't have to worry about it if our node doesn't exist. Since this function is called by the Termination Controller's `Reconcile()` function, it will attempt to reconcile only nodes that exist in the cluster. 

If a node were to not exist, or get removed while we are doing this work, it will return an error and attempt to re-queue up a reconcile request. Once that reconcile request for this node is processed, it will realize there is no node, and then cease work.